### ISSUE TITLE
Agent should support Java 7

### ISSUE THREAD

**Issue by karottenreibe** [cqse/teamscale-jacoco-agent#40]
Currently the Agent requires Java 8 to run. We should support Java 7 as well.

**Comment by salsolatragus** [comment]
We have a customer where Java 7 is still in use, such that we cannot use the agent. Is there any reason for the Java 8 requirement other than the usage of Java 8 features (lambdas, streams, ...) in our own source code?

**Comment by karottenreibe** [comment]
@salsolatragus I just discussed this with @agoeb: The customer is fine with not having Java 7 support and will use plain JaCoCo. We did an impact analysis (i.e. @DreierF converted the whole thing to Kotlin ;-)) and there would be significant effort to implement this. Given that we have rather limited resources ATM, I just don't see this as a priority, so we won't implement this until we have more "spare time".

The challenges we faced were a) as you mentioned Java 8 features used in our code and b) dependencies that require Java 8. So it's not as trivial as it looks at first glance, unfortunately.

I'll close this issue for now so noone works on this accidentally. I'd like to be involved if anyone wants to pick this up at a later, more convenient time.
